File: internal/types/artifact_version_pull.go
package types

import "time"

type ArtifactVersionPull struct {
	CreatedAt            time.Time             `json:"createdAt"`
	RemoteAddress        *string               `json:"remoteAddress,omitempty"`
	UserAccount          *UserAccount          `json:"userAccount,omitempty"`
	CustomerOrganization *CustomerOrganization `json:"customerOrganization,omitempty"`
	Artifact             Artifact              `json:"artifact"`
	ArtifactVersion      ArtifactVersion       `json:"artifactVersion"`
}

File: frontend/ui/src/app/types/artifact-version-pull.ts
import {BaseArtifact, BaseArtifactVersion} from '../services/artifacts.service';

export interface ArtifactVersionPull {
  createdAt: string;
  remoteAddress?: string;
  userAccountName?: string;
  userAccountEmail?: string;
  customerOrganizationName?: string;
  artifact: BaseArtifact;
  artifactVersion: BaseArtifactVersion;
}
